### PR TITLE
External DNS Operator release notes 4.11

### PR DESCRIPTION
**Pull request by xenolinux**
Version(s): OCP 4.11

Issue:

Link to docs preview: http://file.rdu.redhat.com/sdudhgao/EDNS-RN-411/release_notes/ocp-4-11-release-notes.html#ocp-4-11-external-dns-operator

Additional information:

SME: @alebedev87 
QE: @quarterpin

### CODE REVIEW

**Comment by xenolinux** [comment]
> The release note needs a mention of "AWS govcloud support is added as techpreview for this release".

@quarterpin Addressed

File: release_notes/ocp-4-11-release-notes.adoc [hunk 1]
```diff
@@ -170,6 +170,11 @@ Previously, you could not specify the subdomain of a route, and the `spec.host`
 
 You can use this enhancement to simplify sharding by enabling a route to have multiple, distinct host names determined by each router deployment that exposes the route.
 
+[id="ocp-4-11-external-dns-operator"]
+==== External DNS Operator is GA for AWS, Azure, and GCP and is TechPreview for AWS GovCloud
```
Comment (alebedev87) [inline release_notes/ocp-4-11-release-notes.adoc:174]:
```suggestion
==== External DNS Operator is GA for AWS Route53, Azure DNS, Google Cloud DNS and Infoblox and is TechPreview for BlueCat and AWS Route53 on GovCloud
```

File: release_notes/ocp-4-11-release-notes.adoc [hunk 2]
```diff
@@ -170,6 +170,11 @@ Previously, you could not specify the subdomain of a route, and the `spec.host`
...
+
+The External DNS Operator is GA for AWS, Azure, GCP and is TechPreview for AWS GovCloud in {product-title} {product-version}. You can now create DNS records on a public DNS zone on Infoblox by using the External DNS Operator. Previously, you needed to create multiple namespaces and roles to install the External DNS Operator. You can now use a single namespace `external-dns`. You must migrate from TechPreview to GA. For more information, see xref:../networking/external_dns_operator/nw-installing-external-dns-operator-on-cloud-providers.adoc#nw-installing-external-dns-operator-on-cloud-providers[About the External DNS Operator].
```
Comment (alebedev87) [inline release_notes/ocp-4-11-release-notes.adoc:176]:
```suggestion
The External DNS Operator is GA for AWS, Azure, GCP and is TechPreview for AWS GovCloud in {product-title} {product-version}. You can now create DNS records on a public DNS zone on Infoblox by using the External DNS Operator. Previously, you needed to create a namespace for the operands and some RBAC prior the installation of the External DNS Operator. External DNS Operator deploys the operands in its own namespace: `external-dns-operator` (by default). You must migrate from TechPreview to GA. For more information, see xref:../networking/external_dns_operator/nw-installing-external-dns-operator-on-cloud-providers.adoc#nw-installing-external-dns-operator-on-cloud-providers[About the External DNS Operator].
```

File: release_notes/ocp-4-11-release-notes.adoc [hunk 3]
```diff
@@ -170,6 +170,11 @@ Previously, you could not specify the subdomain of a route, and the `spec.host`
...
+
```
Comment (alebedev87) [inline release_notes/ocp-4-11-release-notes.adoc:177]: This link can be added as a migration guide: https://github.com/openshift/external-dns-operator/blob/main/docs/migration-guides.md#from-01x-techpreview-to-10x-ga

**Comment by alebedev87** [comment]
@xenolinux: We need to add some words about the upstream version used for this release: it'll be `v0.12.0`, techpreview was `v0.10.2`.

File: release_notes/ocp-4-11-release-notes.adoc (hunk 3 above)
Comment (xenolinux) [reply release_notes/ocp-4-11-release-notes.adoc:177]: Addressed

File: release_notes/ocp-4-11-release-notes.adoc (hunk 2 above)
Comment (xenolinux) [reply release_notes/ocp-4-11-release-notes.adoc]: Addressed

File: release_notes/ocp-4-11-release-notes.adoc (hunk 1 above)
Comment (xenolinux) [reply release_notes/ocp-4-11-release-notes.adoc]: to make the heading shorter and crisp, I added this information in the paragraph and changed the heading to  "External DNS Operator is GA in OpenShift Container Platfrom 4.11"

File: release_notes/ocp-4-11-release-notes.adoc (hunk 3 above)
Comment (xenolinux) [reply release_notes/ocp-4-11-release-notes.adoc:177]: hmm, As per the doc guidelines we cannot link to a GitHub repo.

**Comment by xenolinux** [comment]
> v0.10.2

Addressed

**Comment by xenolinux** [comment]
> @xenolinux: We need to add some words about the upstream version used for this release: it'll be `v0.12.0`, techpreview was `v0.10.2`.

Addressed

File: release_notes/ocp-4-11-release-notes.adoc [hunk 4]
```diff
@@ -170,6 +170,11 @@ Previously, you could not specify the subdomain of a route, and the `spec.host`
 
 You can use this enhancement to simplify sharding by enabling a route to have multiple, distinct host names determined by each router deployment that exposes the route.
 
+[id="ocp-4-11-external-dns-operator"]
+==== External DNS Operator is GA in {product-title} 4.11
+
+The External DNS Operator is GA for AWS Route53, Azure DNS, GCP DNS, and Infoblox. The External DNS Operator is TechPreview for BlueCat and AWS Route53 on GovCloud. You can now create DNS records on a public DNS zone on Infoblox by using the External DNS Operator. Previously, you needed to create a namespace for the operands and some RBAC prior the installation of the External DNS Operator. By default, the External DNS Operator deploys the operands in the namespace `external-dns-operator`. You must migrate from TechPreview to GA. The upstream version for an {product-title} 4.11 is `v0.12.0` and for the TechPreview was `v0.10.2`. For more information, see xref:../networking/external_dns_operator/nw-installing-external-dns-operator-on-cloud-providers.adoc#nw-installing-external-dns-operator-on-cloud-providers[About the External DNS Operator].
```
Comment (alebedev87) [inline release_notes/ocp-4-11-release-notes.adoc:176]: Sorry forgot about that but we need to add some details on the changes of the upstream version. So the new version contains the following main changes:

- Status of the routes source is now used to retrieve the DNS FQDN names (useful in regard of the route subdomain feature)
- BlueCat DNS provider now allows the users to enable the automatic DNS config deployment
- Proxy support for BlueCat DNS provider

Comment (xenolinux) [reply release_notes/ocp-4-11-release-notes.adoc]: Addressed

File: release_notes/ocp-4-11-release-notes.adoc [hunk 5]
```diff
@@ -170,6 +170,11 @@ Previously, you could not specify the subdomain of a route, and the `spec.host`
...
 
+[id="ocp-4-11-external-dns-operator"]
+==== External DNS Operator is GA in {product-title} 4.11
+
+The External DNS Operator is GA for AWS Route53, Azure DNS, GCP DNS, and Infoblox. The External DNS Operator is TechPreview for BlueCat and AWS Route53 on GovCloud. You can now create DNS records on a public DNS zone on Infoblox by using the External DNS Operator. Previously, you needed to create a namespace for the operands and some RBAC prior the installation of the External DNS Operator. By default, the External DNS Operator deploys the operands in the namespace `external-dns-operator`. You must migrate from TechPreview to GA. The upstream version for an {product-title} 4.11 is `v0.12.0` and for the TechPreview was `v0.10.2`. You can use the status of the routes source to retrieve the DNS FQDN names. {product-title} 4.11 adds the proxy support for BlueCat DNS provider. You can enable the automatic DNS configuration deployment by using the BlueCat DNS provider. For more information, see xref:../networking/external_dns_operator/nw-installing-external-dns-operator-on-cloud-providers.adoc#nw-installing-external-dns-operator-on-cloud-providers[About the External DNS Operator].
```
Comment (alebedev87) [inline release_notes/ocp-4-11-release-notes.adoc:176]:
```suggestion
The External DNS Operator is GA for AWS Route53, Azure DNS, GCP DNS, and Infoblox. The External DNS Operator is TechPreview for BlueCat and AWS Route53 on GovCloud. You can now create DNS records on a public DNS zone on Infoblox by using the External DNS Operator. Previously, you needed to create a namespace for the operands and some RBAC prior the installation of the External DNS Operator. By default, the External DNS Operator deploys the operands in the namespace `external-dns-operator`. You must migrate from TechPreview to GA. The upstream version for an {product-title} 4.11 is `v0.12.0` and for the TechPreview was `v0.10.2`. The route's status will now be used to retrieve the DNS FQDN names.  The proxy support for BlueCat DNS provider is now added. You can enable the automatic DNS configuration deployment while the BlueCat DNS provider. For more information, see xref:../networking/external_dns_operator/nw-installing-external-dns-operator-on-cloud-providers.adoc#nw-installing-external-dns-operator-on-cloud-providers[About the External DNS Operator].
```

Comment (xenolinux) [reply release_notes/ocp-4-11-release-notes.adoc]: Addressed

File: release_notes/ocp-4-11-release-notes.adoc (hunk 3 above)
Comment (xenolinux) [reply release_notes/ocp-4-11-release-notes.adoc:177]: I am checking how we can address this migration guide approach.

File: release_notes/ocp-4-11-release-notes.adoc [hunk 6]
```diff
@@ -170,6 +170,11 @@ Previously, you could not specify the subdomain of a route, and the `spec.host`
...
 
+[id="ocp-4-11-external-dns-operator"]
+==== External DNS Operator is GA in {product-title} 4.11
+
+The External DNS Operator is GA for AWS Route53, Azure DNS, GCP DNS, and Infoblox. The External DNS Operator is TechPreview for BlueCat and AWS Route53 on GovCloud. You can now create DNS records on a public DNS zone on Infoblox by using the External DNS Operator. Previously, you needed to create a namespace for the operands and some RBAC prior the installation of the External DNS Operator. By default, the External DNS Operator deploys the operands in the namespace `external-dns-operator`. You must migrate from TechPreview to GA. The upstream version for an {product-title} 4.11 is `v0.12.0` and for the TechPreview was `v0.10.2`. You can now use the status of the route to retrieve the DNS FQDN names. The proxy support for BlueCat DNS provider is now added. You can enable the automatic DNS configuration deployment while the BlueCat DNS provider. For more information, see xref:../networking/external_dns_operator/nw-installing-external-dns-operator-on-cloud-providers.adoc#nw-installing-external-dns-operator-on-cloud-providers[About the External DNS Operator].
```
Comment (darshan-nagaraj) [inline release_notes/ocp-4-11-release-notes.adoc:176]:
```suggestion
In {product-title} 4.11, the External DNS Operator is available for AWS Route53, Azure DNS, GCP DNS, and Infoblox as a General Availability (GA). The External DNS Operator is still a Technology Preview (TP) for BlueCat and AWS Route53 on GovCloud. 

With this update, the External DNS Operator provides the following enhancements:

* You can create DNS records on a public DNS zone on Infoblox. 
* External DNS Operator deploys the operands in the namespace `external-dns-operator`. 
* You can use the status of the route to retrieve the DNS FQDN names. 
* The proxy support for BlueCat DNS provider is now available. 
* You can enable the automatic DNS configuration deployment while using the BlueCat DNS provider. 

Ensure that you must migrate from TP to GA. The upstream version for an {product-title} 4.11 is `v0.12.0` and for the TP is `v0.10.2`. 

For more information, see xref:../networking/external_dns_operator/nw-installing-external-dns-operator-on-cloud-providers.adoc#nw-installing-external-dns-operator-on-cloud-providers[About the External DNS Operator].

```

**Review by darshan-nagaraj — COMMENTED** [review]
I have left my comments. PTAL

File: release_notes/ocp-4-11-release-notes.adoc (hunk 6 above)
Comment (xenolinux) [reply release_notes/ocp-4-11-release-notes.adoc]: Addressed

File: release_notes/ocp-4-11-release-notes.adoc [hunk 7]
```diff
@@ -170,6 +170,11 @@ Previously, you could not specify the subdomain of a route, and the `spec.host`
 
 You can use this enhancement to simplify sharding by enabling a route to have multiple, distinct host names determined by each router deployment that exposes the route.
 
+[id="ocp-4-11-external-dns-operator"]
+==== External DNS Operator is GA in {product-title} 4.11
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply release_notes/ocp-4-11-release-notes.adoc]: Addressed

File: release_notes/ocp-4-11-release-notes.adoc [hunk 8]
```diff
@@ -170,6 +170,21 @@ Previously, you could not specify the subdomain of a route, and the `spec.host`
 
 You can use this enhancement to simplify sharding by enabling a route to have multiple, distinct host names determined by each router deployment that exposes the route.
 
+[id="ocp-4-11-external-dns-operator"]
+==== External DNS Operator
+
+In {product-title} 4.11, the External DNS Operator is available for AWS Route53, Azure DNS, GCP DNS, and Infoblox as a General Availability (GA). The External DNS Operator is still a Technology Preview (TP) for BlueCat and AWS Route53 on GovCloud.
```
Comment (maxwelldb) [inline release_notes/ocp-4-11-release-notes.adoc:176]: This should be rephrased. The Operator is not a General Availability, but it has reached GA status. Likewise, it is not a Technology Preview, but it is in TP status.

**Review by maxwelldb — COMMENTED** [review]
Left a couple of suggestions.

File: release_notes/ocp-4-11-release-notes.adoc (hunk 8 above)
Comment (xenolinux) [reply release_notes/ocp-4-11-release-notes.adoc]: Addressed

File: release_notes/ocp-4-11-release-notes.adoc [hunk 9]
```diff
@@ -170,6 +170,21 @@ Previously, you could not specify the subdomain of a route, and the `spec.host`
...
+With this update, the External DNS Operator provides the following enhancements:
+
+* You can create DNS records on a DNS zone on Infoblox.
+* You do not have to manually create a namespace for the operands and Role-based access control (RBAC) prior to installation. By default, the External DNS Operator creates the operands in the namespace `external-dns-operator`.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply release_notes/ocp-4-11-release-notes.adoc]: Addressed

File: release_notes/ocp-4-11-release-notes.adoc [hunk 10]
```diff
@@ -170,6 +170,21 @@ Previously, you could not specify the subdomain of a route, and the `spec.host`
 
 You can use this enhancement to simplify sharding by enabling a route to have multiple, distinct host names determined by each router deployment that exposes the route.
 
+[id="ocp-4-11-external-dns-operator"]
+==== External DNS Operator
+
+In {product-title} 4.11, the External DNS Operator is available for AWS Route53, Azure DNS, GCP DNS, and Infoblox in a General Availability (GA) status. The External DNS Operator is still in a Technology Preview (TP) status for BlueCat and AWS Route53 on GovCloud.
```
Comment (maxwelldb) [inline release_notes/ocp-4-11-release-notes.adoc:176]:
```suggestion
In {product-title} 4.11, the External DNS Operator is available for AWS Route53, Azure DNS, GCP DNS, and Infoblox in General Availability (GA) status. The External DNS Operator is still in Technology Preview (TP) status for BlueCat and AWS Route53 on GovCloud.
```

**Review by maxwelldb — APPROVED** [review]

**Pull request merged**